src/export/ValueSetExporter.ts

```diff
@@ -76,6 +76,30 @@ export class ValueSetExporter {
             .replace(/^([^|]+)/, csMetadata?.url ?? '$1')
             .split('|');
           composeElement.system = foundSystem[0];
+          // if the code system is also a contained resource, add the valueset-system extension
+          // this zulip thread contains a discussion of the issue and an example using this extension:
+          // https://chat.fhir.org/#narrow/stream/215610-shorthand/topic/Contained.20code.20system.20in.20the.20value.20set/near/424938537
+          // additionally, if it's not a contained resource, and the system we found is an inline instance, that's a problem
+          const containedSystem = valueSet.contained?.find((resource: any) => {
+            return resource?.id === csMetadata.id && resource.resourceType === 'CodeSystem';
```

Comment:
Part of #1402 says

> We should also check to see if you can refer to inline value sets from compose rules (and fix it if you can't).

I went to go look to see if this was fixed with this PR, so I tried to add an inline valueset to the contained list and reference it in a compose rule. Maybe it is worth discussing if we want that fix as part of this or if we want to track it as part of a separate issue. Either way, we probably should fix this line because I get an error that says "Cannot read properties of undefined (reading 'id')". So at the very least, I think we want to avoid that. I think updating the line like this should do it:
```suggestion
            return resource?.id === csMetadata?.id && resource.resourceType === 'CodeSystem';
```